### PR TITLE
🌱Add CAPI E2E tests

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:
Adding CAPI tests except KCP adoption.
MachinePool test will be uncommented once https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2137 is done.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2132

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -90,10 +90,12 @@ PULL_POLICY ?= Always
 LDFLAGS := $(shell source ./hack/version.sh; version::ldflags)
 
 # 'functional tests' as the ginkgo filter will run ALL tests ~ 2 hours @ 3 node concurrency.
-E2E_UNMANAGED_FOCUS ?= "functional tests - unmanaged"
+E2E_UNMANAGED_FOCUS ?= "Cluster API E2E tests"
```
Comment (randomvariable) [inline Makefile:93]: This is essentially disabling the functional tests in favour of the E2E tests?

Comment (sedefsavas) [reply Makefile:93]: Actually I was going to run these tests by setting the focus in test-infra for both AWS specific periodic e2e and CAPI e2e.
But you are right, I will revert it to normal, then only enable CAPI e2e in the new periodic infra test.

Comment (sedefsavas) [reply Makefile:93]: Ok, didn't see this gap analysis. Taking a look.

**Comment by randomvariable** [comment]
short term, can we set this up as a new periodic in test-infra ?

**Comment by sedefsavas** [comment]
/hold

@randomvariable yes, will do. 

Just testing MachinePool test for now. Will change the focus after that.

**Comment by sedefsavas** [comment]
/test pull-cluster-api-provider-aws-e2e
--

**Comment by randomvariable** [comment]
need both
`make generate`
and linting

**Comment by sedefsavas** [comment]
/hold
until https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2197, 2198, 2194, 2193 to get merged.

**Comment by sedefsavas** [comment]
TODO:
Revert `E2E_UNMANAGED_FOCUS` after e2e tests pass.

**Comment by sedefsavas** [comment]
This PR is ready, can be merged after https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2193 is in.
This does not affect any existing e2e tests, we created a new periodic job that runs this: https://github.com/kubernetes/test-infra/pull/20463

**Comment by sedefsavas** [comment]
Test fails due to timeout:

`{"component":"entrypoint","file":"prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 5h0m0s timeout","severity":"error","time":"2021-01-15T01:32:59Z"}
`

**Comment by randomvariable** [comment]
Let's merge this and the test infra one on Monday, so we don't get a bunch of emails over the weekend if it's there's issues.

Please remind me.

**Comment by sedefsavas** [comment]
@randomvariable Shall we merge this?

**Comment by sedefsavas** [comment]
I think we can merge this.
e2e test failed due to timeout, those tests are not changed in this PR.

cc @richardcase @randomvariable

**Pull request merged**